### PR TITLE
feat: added config for adaptive instance scaling

### PR DESCRIPTION
**Pull request by klyse**

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -166,6 +166,10 @@ session_management_cors_origin:
 session_management_oauth_client_id:
 session_management_oauth_client_secret:
 session_management_ip_stack_api_token: "{{ cloudxr_configuration.ip_stack_api_token }}"
+session_management_adaptive_instance_scaling_enabled: 'true'
```
Comment (burnedikt) [inline defaults/main.yml:169]: to be consistent throughout this role, we should actually use boolean values for boolean values and then convert them to strings when needed (e.g. if used as an environment variable).

```suggestion
session_management_adaptive_instance_scaling_enabled: true
```

File: tasks/session_management_service.yml [hunk 2]
```diff
@@ -49,6 +49,10 @@
       IpLookupService__ApiToken: "{{ session_management_configuration.ip_stack_api_token }}"
       CloudXRManagement__BaseUrl: "http://{{ container_names.cloudxr_management }}:5000"
       Sentry__Dsn: "{{ session_management_configuration.sentry_dsn }}"
+      AdaptiveInstanceScaling__Enabled: "{{ session_management_configuration.adaptive_instance_scaling.enabled }}"
```
Comment (burnedikt) [inline tasks/session_management_service.yml:52]:
```suggestion
      AdaptiveInstanceScaling__Enabled: "{{ session_management_configuration.adaptive_instance_scaling.enabled | ternary('true', 'false') }}"
```

see https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#defining-different-values-for-true-false-null-ternary

File: README.md [hunk 3]
```diff
@@ -336,6 +336,26 @@ If custom CORS origin(s) are required it can be set here separated by `,` (comma
...
 
+    session_management_ip_stack_api_token
+
+**Mandatory** The IP address lookup is required to select the closest cloudXR region. The token must be obtained on [this](https://ipstack.com/) website
+
+    session_management_adaptive_instance_scaling_enabled
+
+Enables adaptive instance scaling. This defaults to `True`
+
+    session_management_adaptive_instance_scaling_user_scale_ratio
```
Comment (burnedikt) [inline README.md:347]: documentation is not matching the code. The default is `'true'` (stringified, lowercase version of True). It should be `true` (actual boolean value). Regarding the stringified version, not sure if ASP.NET Core can handle both `'True'` and `'true'`, better double check.

**Review by burnedikt — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (klyse) [reply defaults/main.yml:169]: I tried that already but it creates this error:
image

Comment (klyse) [reply defaults/main.yml:169]: ah, in combination with ternary it makes sense.

**Review by burnedikt — APPROVED** [review]

**Pull request merged**